service/src/test/java/uk/gov/hmcts/reform/fpl/service/ManageDocumentServiceTest.java

```diff
@@ -0,0 +1,532 @@
+    }
+
+    @Test
+    void shouldExpandSupportingEvidenceCollectionWhenEmpty() {
+        List<Element<SupportingEvidenceBundle>> emptySupportingEvidenceCollection = List.of();
+        List<Element<SupportingEvidenceBundle>> supportingEvidenceBundleCollection
+            = manageDocumentService.getSupportingEvidenceBundle(emptySupportingEvidenceCollection);
+
+        assertThat(supportingEvidenceBundleCollection).isNotEmpty();
+    }
+
+    @Test
+    void shouldPersistExistingSupportingEvidenceBundleWhenExists() {
+        List<Element<SupportingEvidenceBundle>> supportEvidenceBundle = buildSupportingEvidenceBundle();
+        List<Element<SupportingEvidenceBundle>> updatedSupportEvidenceBundle =
+            manageDocumentService.getSupportingEvidenceBundle(supportEvidenceBundle);
+
+        assertThat(updatedSupportEvidenceBundle).isEqualTo(supportEvidenceBundle);
+    }
+
+    @Test
+    void shouldReturnEmptyCollectionWhenFurtherEvidenceIsNotRelatedToHearingAndCollectionIsNotPresent() {
+        CaseData caseData = CaseData.builder()
+            .manageDocument(buildManagementDocument(FURTHER_EVIDENCE_DOCUMENTS, NO.getValue()))
+            .build();
+
+        List<Element<SupportingEvidenceBundle>> supportingEvidenceBundleCollection =
+            manageDocumentService.getFurtherEvidenceCollection(caseData);
+
+        assertThat(supportingEvidenceBundleCollection).isNotEmpty();
+        assertThat(supportingEvidenceBundleCollection.get(0).getValue())
+            .isEqualTo(SupportingEvidenceBundle.builder().build());
+    }
+
+    @Test
+    void shouldReturnFurtherEvidenceCollectionWhenFurtherEvidenceIsNotRelatedToHearingAndCollectionIsPresent() {
+        List<Element<SupportingEvidenceBundle>> furtherEvidenceBundle = buildSupportingEvidenceBundle();
+
+        CaseData caseData = CaseData.builder()
+            .manageDocument(buildManagementDocument(FURTHER_EVIDENCE_DOCUMENTS, NO.getValue()))
+            .furtherEvidenceDocuments(furtherEvidenceBundle)
+            .build();
+
+        List<Element<SupportingEvidenceBundle>> furtherDocumentBundleCollection =
+            manageDocumentService.getFurtherEvidenceCollection(caseData);
+
+        assertThat(furtherDocumentBundleCollection).isEqualTo(furtherEvidenceBundle);
+    }
+
+    @Test
+    void shouldReturnHearingEvidenceCollectionWhenFurtherEvidenceIsRelatedToHearingWithExistingEntryInCollection() {
+        List<Element<SupportingEvidenceBundle>> furtherEvidenceBundle = buildSupportingEvidenceBundle();
+        UUID hearingId = UUID.randomUUID();
+        List<Element<HearingBooking>> hearingBookings = List.of(element(hearingId, buildFinalHearingBooking()));
+
+        CaseData caseData = CaseData.builder()
+            .hearingDetails(hearingBookings)
+            .manageDocumentsHearingList(asDynamicList(hearingBookings, hearingId, hearing -> hearing.toLabel(DATE)))
+            .hearingFurtherEvidenceDocuments(List.of(
```

Comment:
Would be helpful to have more hearings with docs maybe to see if approprite one is selected?